servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -1639,6 +1640,51 @@ public DomainRoleMembers listDomainRoleMembers(String domainName) {
         return listReviewRoleMembersWithFilter(domainName, (roleMember -> true), "listDomainRoleMembers");
     }
 
+    @Override
+    public DomainRoleMember getAllRoles(String principal, String domainName) {
+        DomainRoleMember domainRoleMember = new DomainRoleMember();
+        domainRoleMember.setMemberName(principal);
+        domainRoleMember.setMemberRoles(new ArrayList<>());
+
+        if (!StringUtil.isEmpty(domainName)) {
+            getDomainRolesForPrincipal(principal, domainRoleMember, domainName);
+        } else {
+            String[] fnames = getDomainList();
+            for (String domain : fnames) {
+                getDomainRolesForPrincipal(principal, domainRoleMember, domain);
+            }
+        }
+
+        return domainRoleMember;
+    }
+
+    private void getDomainRolesForPrincipal(String principal, DomainRoleMember domainRoleMember, String domain) {
+        DomainStruct dom = getDomainStruct(domain);
+        if (dom == null) {
+            return;
+        }
+
+        for (Role role: dom.getRoles().values()) {
+            List<RoleMember> roleMembers = role.getRoleMembers();
+            if (roleMembers == null) {
+                continue;
+            }
+            for (RoleMember roleMember : roleMembers) {
+                if (!roleMember.getMemberName().equals(principal)) {
+                    continue;
+                }
+
+                MemberRole memberRole = new MemberRole();
```

Comment:
we should also return the system disabled flag if we're return all the other attributes

Reply:
Done